examples/lighting-app/silabs/src/LightingManager.cpp

```diff
@@ -307,3 +363,72 @@ void LightingManager::OnTriggerOffWithEffect(OnOffEffect * effect)
     LightMgr().mOffEffectArmed = true;
     LightMgr().StartTimer(offEffectDuration);
 }
+
+#if (defined(SL_MATTER_RGB_LED_ENABLED) && SL_MATTER_RGB_LED_ENABLED == 1)
+bool LightingManager::InitiateLightctrlAction(int32_t aActor, Action_t aAction, uint16_t size, uint32_t aAttributeId,
+                                              uint8_t * value)
+{
+    bool action_initiated = false;
+    VerifyOrReturnError(aAction == COLOR_ACTION_XY || aAction == COLOR_ACTION_HSV || aAction == COLOR_ACTION_CT, action_initiated);
+    RGBLEDWidget::ColorData_t colorData;
+    switch (aAction)
+    {
+    case COLOR_ACTION_XY:
+        colorData.xy = {};
+
+        // Check if CurrentX or CurrentY has changed
+        if (aAttributeId == ColorControl::Attributes::CurrentX::Id)
+        {
+            if (mCurrentX == *reinterpret_cast<uint16_t *>(value))
```

Comment:
you can cleanup those 2nd if by using `VerifyOrReturnValue` macro

Reply:
Modified using VerifyOrReturnValue.